bundles/org.openhab.voice.googlestt/src/main/java/org/openhab/voice/googlestt/internal/GoogleSTTService.java

```diff
@@ -272,16 +274,18 @@ && isExpiredInterval(maxSilenceMillis, responseObserver.getLastInputTime())) {
                 logger.debug("Stops listening, max silence time reached");
                 break;
             }
-            if (dataN != readBytes) {
+            if (numBytesRead != remaining) {
+                remaining = remaining - numBytesRead;
                 try {
                     Thread.sleep(100);
```

Comment:
Is this actually needed?

Reply:
Yes, we need to update the remaining so next read just add the missing bytes to the buffer in the correct offset.

Reply:
Yes, we need to update the remaining so next read just add the missing bytes to the correct buffer offset.

Reply:
I was referring to the thread sleep. 🙂 The `read()` itself is blocking.

Reply:
Ups,sorry. Probably not, it was a sugestión on the previous review to prevent damaging the cpu in case the stream start returning 0 bytes forever which I think can not happen but also I think is not breaking anything as it's so small. Also after see the logs the other day I saw we stop there each 4 or 5 reads, so maybe it's already reducing the cpu usage a little.
But really I'm not sure whether or not is a good a idea to have it.

Reply:
If this is the `InputStream` implementation, it should always read at least 1 byte, unless it is requested to read 0 (`len` = 0), but this would be a bug:
https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/io/InputStream.html#read(byte%5B%5D,int,int)

Reply:
Thank you for the reference.
Yes seems quite clear. I didn't read this one before.

So probably better remove it as you said, in case it causes some unwanted behaviour. 

I should remove it in the rustpotter one also.